pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -159,23 +204,29 @@ stdenv.mkDerivation rec {
     "-Donnxruntime_BUILD_UNIT_TESTS=ON"
     "-Donnxruntime_ENABLE_LTO=ON"
     "-Donnxruntime_USE_FULL_PROTOBUF=OFF"
+    (lib.cmakeBool "onnxruntime_USE_CUDA" cudaSupport)
+    (lib.cmakeBool "onnxruntime_USE_NCCL" cudaSupport)
   ] ++ lib.optionals pythonSupport [
     "-Donnxruntime_ENABLE_PYTHON=ON"
+  ] ++ lib.optionals cudaSupport [
+    "-DFETCHCONTENT_SOURCE_DIR_CUTLASS=${cutlass}"
+    "-Donnxruntime_CUDNN_HOME=${cudaPackages.cudnn}"
+    "-DCMAKE_CUDA_ARCHITECTURES=${cudaArchitecturesString}"
   ];
 
-  env = lib.optionalAttrs stdenv.cc.isClang {
+  env = lib.optionalAttrs effectiveStdenv.cc.isClang {
     NIX_CFLAGS_COMPILE = toString [
       "-Wno-error=deprecated-declarations"
       "-Wno-error=unused-but-set-variable"
     ];
   };
 
-  doCheck = true;
+  doCheck = !cudaSupport;
 
```

Comment:
```suggestion

requiredSystemFeatures = lib.optionals cudaSupport [ "big-parallel" ];

```

Reply:
Thank you, that seems reasonable at this point.